Sensors/SMCDellSensors/KeyImplementations.cpp

```diff
@@ -0,0 +1,205 @@
+//
+//  KeyImplementations.cpp
+//  SMCDellSensors
+//
+//  Copyright © 2020 lsv1974. All rights reserved.
+//
+
+#include "KeyImplementations.hpp"
+#include "SMCDellSensors.hpp"
+
+static inline UInt16 swap_value(UInt16 value)
+{
+	return ((value & 0xff00) >> 8) | ((value & 0xff) << 8);
+}
+static inline UInt16 encode_fpe2(UInt16 value) {
+	return swap_value(value << 2);
+}
+static inline UInt16 decode_fpe2(UInt16 value) {
+	return (swap_value(value) >> 2);
+}
+
+SMC_RESULT F0Ac::readAccess() {
+	IOSimpleLockLock(SMIMonitor::getShared()->stateLock);
```

Comment:
Can this and the other accessors be replaced with _Atomic(uint6_t)?

Reply:
I got error: Illegal initializer type '_Atomic(uint16_t)'.

Reply:
Yes, initialization of _Atomic(T) is a bit messy. You can use default initializer if it is a class member, see https://github.com/acidanthera/Lilu/blob/master/Lilu/Headers/kern_util.hpp#L602 for example. Otherwise, you can explicitly write to it in constructor.

The question to answer is whether you really need the mutual exclusion, or just the atomic access to the variables.

Reply:
It has been done in the latest commit, thank you for suggestion!